test/generator/service_test.exs

```diff
@@ -341,23 +363,30 @@ defmodule Thrift.Generator.ServiceTest do
       :ok ->
         :ok
     end
-
+    Process.flag(:trap_exit, true)
     Process.unlink(ctx.server)
     Process.exit(ctx.server, :kill)
-    assert_receive {:DOWN, ^ref, _, _, _}
 
-    assert :sys.get_state(ctx.client).mod_state.retries > 0
+    assert_receive {:DOWN, ^ref, _, _, _}
+    assert_receive {:EXIT, _, {:error, :econnrefused}}
```

Comment:
Who does the :EXIT come from, ctx.client? It would improve the clarity a bit to match on that.